English fix

andi-makes opened the pull request:
Fixed/Improved English in Files:

- [X]  Home.md
- [X]  _Footer.md
- [X]  usage/Installation.md

Todo:

- [x] usage/Frequent-Issues.md
- [x] usage/Using-the-CLI.md
- [x] usage/Using-the-GUI.md

Feedback is highly appreciated!

Commit:
Fixed typos in Home.md

Commit:
Improved _Footer.md

Commit:
Fixed typos and improved usage/Installation.md

Commit:
Merge branch 'master' of https://github.com/codeoverflow-org/chatoverflow-wiki into english_fix

J0B10 commented on usage/Installation.md:
```diff
@@ -1,51 +1,56 @@
 1. Clone the main repository. Then clone the api into folder `codeoverflow/api`, the gui into folder `codeoverflow/gui` and public plugins repository into `codeoverflow/plugins-public`.
 2. Setup the main repository as imported scala project, e.g. using IntelliJ. Make sure to refresh all sbt content and load the custom run configurations.
-3. You're done. Happy coding!
+3. You're done. Happy coding! -->
```
~Why have you commented out the Pro section?~

andi-makes replied:
I think it could be confusing for users who think that they know sbt, scala, etc. and then meet the rather complex structure of the project. 

We can add that section back, no problem. But then I would rename it to something like "Setup for advanced users" or something like that.

J0B10 commented:
It's looking good till now. Ignore my comment on the Installation.md.

We'll propably completly redo the Installation.md to match with the new install scripts from codeoverflow-org/chatoverflow#63

Commit:
Fixed typos and improved usage/Frequent-Issues.md

Commit:
Fixed typos and improved usage/Frequent-Issues.md

Commit:
Merge branch 'english_fix' of https://github.com/DragonCoder01/chatoverflow-wiki into english_fix

Commit:
Merges are difficult :C

Commit:
Fixed typos and improved usage/Installation.md

Commit:
Fixed typos and improved usage/Using-the-CLI.md

Commit:
Fixed typos and improved usage/Using-the-GUI.md

andi-makes commented:
I have decided to only change the files in the usage Folder as well as in the root folder, because I have been struggeling with merging the new changes from master into this pull request. 

I will edit the other pages using the pen, I think this will spare me some headaches 😄 

I've also fixed typos in the new Installation.md. I also made changes to the structure of the file, if there is a need to change it back I of course will do it.

As always, feedback is highly appreciated!

Commit:
Fixed a typo in ...
development/Setting-up-the-Build-Environment.md

Commit:
Fixed typos in ...
development/Adding-a-new-platform-source.md

Commit:
Fixed typos in ...
development/Deploy-ChatOverflow.md

Commit:
Fixed typos and improved ...
development/Writing-a-plugin.md

Commit:
Fixed typos in ...
services/Discord.md

Commit:
Fixed typos in ...
services/Serial.md

Commit:
Fixed typos in ...
services/Twitch-Chat.md

andi-makes commented:
Well, now I have sacrificed my evening with quickly going through all the other files in the wiki thanks to MelanXs message in the Discord channel. All the obvious typos should be gone now, but of course there could be some spelling/grammar wrong.

If there is anything I've missed, let me know, because you know, feedback is highly appreciated!

hlxid commented on development/Writing-a-plugin.md:
````diff
@@ -54,24 +54,24 @@ class MySuperCoolPlugin(manager: PluginManager) extends PluginImpl(manager) {
 ```
 
-Let's talk about *Requirements* first: These are the way to retrieve data from the outside world and returning information to it. Basicaly, you can require everything from a single paramter, set by the user to full access to a users livestream chat.
+Let's talk about *Requirements* first: These are used to retrieve data from the outside world and returning information to it. Basically, you can require everything from a single paramter set by the user to full access to a users livestream chat.
````
Small typo here :wink:
```suggestion
Let's talk about *Requirements* first: These are used to retrieve data from the outside world and returning information to it. Basically, you can require everything from a single parameter set by the user to full access to a users livestream chat.
```

hlxid commented on development/Writing-a-plugin.md:
````diff
@@ -8,15 +8,15 @@ There are 3 important steps when you want to create and implement a new chat ove
+The first class should extend the Pluggable-Interface from `org.codeoverflow.chatoverflow.api.plugin.Pluggable`. When the framework starts up, your plugin will be looked through for a class implementing this interface to get an entry point to your work. A raw version might look like this:
 
  ```
 class MySuperCoolPlug extends Pluggable{
````
I think when we are already on it we can also fix this formatting :smiley:
```suggestion
class MySuperCoolPlug extends Pluggable {
```

hlxid commented on usage/Using-the-GUI.md:
```diff
@@ -1,45 +1,45 @@
+Before running a plugin you have to create a plugin instance.  
 All available plugins are listed under `Plugin types`.  
-Add more plugins by putting the plugin Jar files into the plugins folder.
+Add more plugins by putting the plugin `.jar` files into the plugins folder.
```
I think that we should mention here that the framework has to be restarted to load newly added plugins.

andi-makes replied:
Add your suggestion in [this commit](https://github.com/codeoverflow-org/chatoverflow-wiki/pull/5/commits/b2661939f9132749d529e4830925ef3bb7f980f1).

hlxid commented on usage/Using-the-CLI.md:
````diff
@@ -34,10 +34,10 @@ You can also configure the paths of config- and credentials-files to create cust
 ```
 
-By default, plugin output on console is disable. But you can still enable it with the flag `-o`.
+By default the output of a plugin is disabled. You can enable it with the flag `-o`.
````
We should mention that the output is only disabled for the console, not the gui, to prevent confusion.
```suggestion
By default the log output of a plugin on the console is disabled. You can enable it with the flag `-o`.
```

DragonCoder01 replied:
Added your suggestion in [this commit](https://github.com/codeoverflow-org/chatoverflow-wiki/pull/5/commits/b2661939f9132749d529e4830925ef3bb7f980f1).

hlxid reviewed (commented):
:+1: Looks good to me for the most part (see my comments down below), but because my English isn't that good either I think @joblo2213 should still take a look.

Many thanks for this pr in advance :smiley: !

Commit:
Update development/Writing-a-plugin.md

Co-Authored-By: Daniel Huber <daniel.hub@outlook.de>

Commit:
Update development/Writing-a-plugin.md

Co-Authored-By: Daniel Huber <daniel.hub@outlook.de>

Commit:
Improved usage/Using-the-GUI.md

Added some suggestion from daniel0611.

Commit:
Update usage/Using-the-CLI.md

Co-Authored-By: Daniel Huber <daniel.hub@outlook.de>

justgerd commented on development/Adding-a-new-platform-source.md:
```diff
@@ -1,17 +1,17 @@
-Thank you for help for the development of *Chat Overflow*.  
-While plugins live in their own project, source connections (e.g. to platform like Twitch or Discord) exist in the framework itself.  
-By this, we can make sure that everyone profits from the new possibilites.  
+Thank you for helping developing *Chat Overflow*.  
```
Still sounds a bit weird, what about "thank you for your help in chat overflow development"

rad4day replied:
What would you think of "Thank you for improving *Chat Overflow*"?

justgerd commented on development/Adding-a-new-platform-source.md:
```diff
@@ -1,17 +1,17 @@
-By this, we can make sure that everyone profits from the new possibilites.  
+Thank you for helping developing *Chat Overflow*.  
+While plugins live in their own project, source connections (e.g. to a platform like Twitch or Discord) exist in the framework itself.  
+By this measure, we can make sure that everyone profits from the same features.  
```
by this measure still sounds weird, what about "Because of this"

rad4day replied:
Line 3 also does not convey the same point as before.

andi-makes replied:
The whole sentence in line 3 is kind of weird. If I understand the original sentence correctly, the platform sources (the things that enable you to connect to the platforms) are baked into the framework itself. And if I remember correctly, this is done in order to prevent that someone develops a proprietary plugin/extension? So by bringing all the platform connectors to the framework, everybody has the same access to them.

I can agree that both the original sentence as well as the new sentence is weird, but I can't make it much better if I don't know for certain what is meant by the original.

J0B10 replied:
Proprietary plugins are explicitly allowed: 
>It's getting even better: The Chat Overflow license allows you to sell your custom plugins, creating new services for other streamers.  

_[README.md](https://github.com/codeoverflow-org/chatoverflow/blob/master/README.md)_

**Let me explain it in german in order to avoid misunderstandings:**

Mit Plugins kann jeder sich mit den verschiedenen Diensten verbinden und unter verwendung der API einfach eigene Chatbots, Abläufe oder ähnliches programmieren.

Die Schnittstellen zu den Diensten (wie Twitch, Discord, ...) müssen fest im Framework verankert sein, sodass jeder Plugin Entwickler einfach darauf zugreifen kann. 
Das erlaubt es das jeder alle Features mit eigenen Plugins nutzen kann.
Gleichzeitig wären Proprietäre Plugins theoretisch möglich.

andi-makes replied:
Also Schnittstellen zu Diensten sind im Framework. Verstanden.
Das ist so, damit jeder Entwickler zugriff hat.
Folglich gibt es keine proprietary Dienst-Schnittstellen, richtig?

Hab ich oben auch mit plugin/extension gemeint, im nachhinein ist das aber echt eine schlechte Wortwahl von mir gewesen. Sorry!

Aber hab ich's jz verstanden?

J0B10 replied:
Exakt 👍

justgerd commented on development/Deploy-ChatOverflow.md:
```diff
@@ -3,4 +3,4 @@ To deploy Chat Overflow and create runnable jar files, follow these steps:
 
-A release of chat overflow is simply a zip archive of the deploy folder, without configs or libraries. Don't forget to update Chat Overflow Framework and API Version before a public release.
\ No newline at end of file
+A release of Chat Overflow is simply a zip archive of the deploy folder, without configs or libraries. Don't forget to update Chat Overflow framework and API version numbers before a public release.
```
there's a "the" missing. `A release of Chat Overflow is simply a zip archive of the deploy folder, without configs or libraries. Don't forget to update *the* Chat Overflow framework and API version numbers before a public release.`

justgerd commented on development/Writing-a-plugin.md:
```diff
@@ -8,18 +8,18 @@ There are 3 important steps when you want to create and implement a new chat ove
 ## Create a new plugin project
 
-Use the *Create Plugin* configuration or start the custom task `sbt create` directly. Enter the basic plugin information in the command promt. This includes the name, version and base plugin folder (e.g. `plugins-public`). The command creates the folder structure of the new plugin and adds a basic build file. Here you can also add custom dependencies.
+Use the *Create Plugin* run configuration or start the custom task `sbt create` directly. Enter the basic plugin information in the command promt. This includes the name, version and base plugin folder (e.g. `plugins-public`). The command creates the folder structure of the new plugin and adds a basic build file. Here you can also add custom dependencies.
```
s/promt/prompt

justgerd commented on development/Writing-a-plugin.md:
```diff
@@ -8,18 +8,18 @@ There are 3 important steps when you want to create and implement a new chat ove
+Use the *Create Plugin* run configuration or start the custom task `sbt create` directly. Enter the basic plugin information in the command promt. This includes the name, version and base plugin folder (e.g. `plugins-public`). The command creates the folder structure of the new plugin and adds a basic build file. Here you can also add custom dependencies.
 
-Next, run the custom task `sbt fetch` to let the framework look for new plugins and update the plugin reference file `plugins.sbt`. If you're using IntelliJ, you can now reload the framework by hand to register the plugin as new and custom project. You should also run `sbt reload` (Yes, both actions have different effects although they shouldn't).
+Next, run the custom task `sbt fetch` to let the framework look for new plugins and update the plugin reference file `plugins.sbt`. If you're using IntelliJ, you can now reload the framework by hand to register the plugin as new as well as a custom project. You should also run `sbt reload` (Yes, both actions have different effects although they shouldn't).
```
s/by hand/manually
`register the plugin as new as well as a custom project` just sound horrible, what about "register the plugin as a new custom project"

justgerd commented on development/Writing-a-plugin.md:
```diff
@@ -8,18 +8,18 @@ There are 3 important steps when you want to create and implement a new chat ove
 Next, add your first class to the source folder of the new plugin. Supported languages are *Scala* and *Java*, but this guide will only show the scala way - development of java plugins is very similar.
 
-The first class should extend the Pluggable-Interface from `org.codeoverflow.chatoverflow.api.plugin.Pluggable`. When the framework starts up, your plugin will be looked trough for a class implementing this interface to get an entry point to your work. A raw version might look like this:
+The first class should extend the Pluggable-Interface from `org.codeoverflow.chatoverflow.api.plugin.Pluggable`. When the framework starts up, your plugin will be looked through for a class implementing this interface to get an entry point to your work. A raw version might look like this:
```
s/Pluggable-Interface/`Pluggable` interface
in the second sentence: "will be looked through" sounds weird, what about "When the framework starts up, it will search for a class implementing this interface in your plugin to get an entry point" (drop the "to your work" part?)

justgerd commented on development/Writing-a-plugin.md:
````diff
@@ -36,11 +36,11 @@ class MySuperCoolPlug extends Pluggable{
  ```
 
-**A short explanation**: The first 3 Methods `getName`, `getAuthor` and `getDescription` are simple: Just a return a String representing the meta information of your plugin. This inclucde it's name (please make sure to not name two plugins develeoped by yourself the same), your name or pseudonym and a brief topic of the plugins purpose.
+**A short explanation**: The first 3 Methods `getName`, `getAuthor` and `getDescription` are simple: They are just returning a String representing the meta information of your plugin. This includes its name (please make sure to not name two plugins develeoped by yourself the same), your name or pseudonym and a brief topic of the plugins purpose.
````
s/the first 3 Methods/The first three methods
s/They are just returning/They just return
"please make sure to not name two plugins devele**o**ped by yourself the same": "please make sure not to name two plugins identically"
s/brief topic/brief summary

justgerd commented on development/Writing-a-plugin.md:
```diff
@@ -36,11 +36,11 @@ class MySuperCoolPlug extends Pluggable{
+**A short explanation**: The first 3 Methods `getName`, `getAuthor` and `getDescription` are simple: They are just returning a String representing the meta information of your plugin. This includes its name (please make sure to not name two plugins develeoped by yourself the same), your name or pseudonym and a brief topic of the plugins purpose.
 
-The next two methods return the number of the API-Version, you developed the plugin with. These are evaluated in the loading process to ensure that your plugin works with the framework version from the user. You can get these numbers from `org.codeoverflow.chatoverflow.api.APIVersion`.
+The next two methods return the number of the API-Version you've developed the plugin with. These are evaluated in the loading process to ensure that your plugin works with the framework version from the user. You can get these numbers from `org.codeoverflow.chatoverflow.api.APIVersion`.
```
s/from the user/that a user has installed

rad4day replied:
"works with the framework version from the user." should probably be replaced with "works with the user's framework version"

justgerd commented on development/Writing-a-plugin.md:
```diff
@@ -36,11 +36,11 @@ class MySuperCoolPlug extends Pluggable{
+The next two methods return the number of the API-Version you've developed the plugin with. These are evaluated in the loading process to ensure that your plugin works with the framework version from the user. You can get these numbers from `org.codeoverflow.chatoverflow.api.APIVersion`.
 
-The last method `createNewPluginInstance` is the interesting one: After the framework checked your meta and version information, it will eventually load your plugin completely. In this method (**and only in this method**), you might return an object that extends the plugin interface. This class is will containt your plugin logic.
+The last method `createNewPluginInstance` is the most interesting one: After the framework checked your meta and version information, it will eventually load your plugin completely. In this method (**and only in this method**), you may return an object that extends the plugin interface. This class will contain your plugin logic.
```
s/meta and version information/metadata *(?)*

justgerd commented on development/Writing-a-plugin.md:
````diff
@@ -54,24 +54,24 @@ class MySuperCoolPlugin(manager: PluginManager) extends PluginImpl(manager) {
 ```
 
-This code requires reading-access to a twitch livestream chat and a parameter ("*a name to say hello to"*). Note: You can obviously not specify, which channel the user might choose - but you can be sure, that you will get a working input. This is ensured by the framework. Please note, that the requirements element already exists, if you extend `PluginImpl`.
+This code requires read-access to a twitch livestream chat and a parameter ("*a name to say hello to"*). Note: You can obviously not specify which channel the user might choose - but you can be sure that you will get a working input. The framework takes care of that. Please note that the requirements element already exists if you extend `PluginImpl`.
````
s/read-access/read access

justgerd commented on development/Writing-a-plugin.md:
```diff
@@ -54,24 +54,24 @@ class MySuperCoolPlugin(manager: PluginManager) extends PluginImpl(manager) {
 
-*One last note: If your testing your plugin the first time, a full reload might be needed. Use the custom task `[Advanced] Full Reload and run ChatOverlfow` to do so. Afters this, you can configure the framework to start your plugin using the [CLI](usage/Using-the-CLI.md)!*
\ No newline at end of file
+*One last note: If you'r testing your plugin the first time, a full reload might be needed. Use the custom task `[Advanced] Full Reload and run ChatOverlfow` to do so. Afters this, you can configure the framework to start your plugin using the [CLI](usage/Using-the-CLI.md)!*
```
s/you'r/you are
s/the first time/for the first time
s/ChatOverlfow/ChatOverflow

justgerd commented on services/Discord.md:
```diff
@@ -1,7 +1,7 @@
-The discord service allows to connect to a discord text channel to get a list of recent messages,
-react on new / edited / deleted messages and reactions.  
+The discord service allows you to connect to a discord text channel to get a list of recent messages,
+react on new / edited / deleted messages as well as reactions.  
```
"react on new / edited / deleted": "react on new, edited or deleted messages"

justgerd commented on services/Serial.md:
```diff
@@ -1,6 +1,6 @@
 <p><img align="right" width="128" height="54" src="/img/services/serial-arduino-logo.png"></p>
 
-The Serial service allows to communicate with a device that is connected to your pc over a serial port.  
+The Serial service allows you to communicate with a device that is connected to your pc over a serial port.  
```
s/pc/PC

justgerd commented on services/Twitch-Chat.md:
```diff
@@ -7,12 +7,12 @@ They can get a List of recent messages, listen for new messages or send messages
 
 ### `oauth` _(required)_ 
-The authentication token for accessing the twitch irc chat with your account.  
+The authentication token for accessing the twitch IRC chat with your account.  
```
s/twitch/Twitch

justgerd commented on usage/Frequent-Issues.md:
```diff
@@ -1,11 +1,11 @@
 ## Run Configuration Bug 
-Due to IntelliJ version differences and usage on Windows vs. Mac OS, the run configurations show problems, sometimes. Change the *classpath or module* under `Edit configurations...` back to the Chat Overflow root module, and everything should be fine.
+Due to IntelliJ version differences as well as differences on platforms (e.g. Windows vs. Mac OS) the run configurations could show problems. If this error occurs, change the `classpath or module` setting under `Edit configurations...` back to the Chat Overflow root module. Now everything should be working fine again.
```
this whole paragraph is confusing to read, not sure if that could be improved though

andi-makes replied:
I think I got it, this is the new version:

Because of the differences in IntelliJ versions the run configurations could be broken. If they are broken, change the `classpath or module` setting in `Edit configurations...` to ChatOverflow root module. Now everything should be working fine again.

I left out the platform differences, because that is part of the confusing part. The end user has a simple explaination for the problem, so that should be fine :D

justgerd commented on usage/Frequent-Issues.md:
```diff
@@ -1,11 +1,11 @@
 If you see this image or error message you are probably running on an old java version that doesn't work with AES 256 bit encryption.  
-Please follow the instructions from the error message and update java to at least `1.8.0_161`.  
-More information about the issue can be found [on stackoverflow.com](https://stackoverflow.com/questions/3862800/invalidkeyexception-illegal-key-size).
+To fix this error you have to update java to at least `1.8.0_161`.  
```
s/java/Java
s/to at least/to version [...] or later

justgerd commented on usage/Installation.md:
```diff
@@ -1,34 +1,23 @@
 
-For running Chatoverflow you need Java 8, version `1.8.0_161` or newer.  
-[Download it form here.](https://www.java.com/en/download/)
+For running Chatoverflow you will need Java 8, version `1.8.0_161` or newer.  
```
s/newer/later *(?)*

justgerd commented on usage/Installation.md:
```diff
@@ -1,34 +1,23 @@
-For running Chatoverflow you need Java 8, version `1.8.0_161` or newer.  
-[Download it form here.](https://www.java.com/en/download/)
+For running Chatoverflow you will need Java 8, version `1.8.0_161` or newer.  
+[You can download it here.](https://www.java.com/en/download/)
```
s/[You can download it here]/You can download it [here]

justgerd commented on usage/Using-the-GUI.md:
```diff
@@ -1,45 +1,45 @@
 When running for the first time enter a password that you want to use for securing your credentials and click on `REGISTER`.  
-If you have already registered, use your password to login.  
-If the login was successful the gui will display the authentication key for that session.  
+If you already are registered, use your password to login.  
```
are needs to come before already

andi-makes replied:
I don't understand? What do you mean?

J0B10 replied:
`If you are already`

justgerd commented on usage/Using-the-GUI.md:
```diff
@@ -1,45 +1,45 @@
+Before running a plugin you have to create a plugin instance.  
 All available plugins are listed under `Plugin types`.  
-Add more plugins by putting the plugin Jar files into the plugins folder.
+Add more plugins by putting the plugin `.jar` files into the plugins folder. If you are adding new plugins to the folder when the framework is running, you need to restart the framework in order to load the newly added plugins.
```
s/when the framework is running/while the framework is running
s/in order to load/in order for it to load

justgerd commented on usage/Using-the-GUI.md:
```diff
@@ -1,45 +1,45 @@
 ![](/img/usage/plugin-types.png)
 
-Enter the plugin name and the author of the plugin and give the instance a name (whatever you like), the click on `CREATE`: 
+Enter the plugin name as well as the author of the plugin and give the instance a name (whatever you like), then click on `CREATE`: 
```
"Enter the plugin and author name [...]"

justgerd commented on usage/Using-the-GUI.md:
```diff
@@ -1,45 +1,45 @@
-You can use the pen to copy an instances name.  
-There are buttons that allow you to `START` / `STOP` a instance, to show its log (if running) or requirements  and a button to `DELETE` the instance (just make sure to stop it before deleting it).
+You can use the pen to copy the name of an instance.  
+There are buttons that allow you to `START` / `STOP` an instance, to show its log (if running) or requirements  and a button to `DELETE` the instance (just make sure to stop it before deleting it).
```
that double space... `requirements  and a button`

justgerd reviewed (changes requested):
Sorry for the gigantic wall of requested changes here, most of them are probably just nitpicks. I have a feeling that I'll have to read through the whole docs at some point to find even more small and insignificant mistakes and make another PR out of it.

J0B10 commented:
It's getting better and better, thanks for all your contribution 😄 👍

Commit:
Implemented all the changes Gerd suggested but ...

Line 3 in development/Adding-a-new-platform-source.md.
A imporvement for it will be in the next commit.

Commit:
Fixed famous line 3 in Adding platform source!

As well as a small change in Using-the-Gui.md

Commit:
Improved Frequent-Issues.md

Added name of the root module
